### PR TITLE
Adding initial data models 

### PR DESCRIPTION
**Pull request by julia326**
... and API endpoint to get all core data joined with batch, state

### CODE REVIEW

File: app/api/data.py [hunk 1]
```diff
@@ -21,6 +21,23 @@ def get_batches():
...
+    states = States.query.all()
+    return jsonify({
+        'states': [state.to_dict() for state in states]
+    })
+
+@api.route('/data/', methods=['GET'])
+def get_core_data():
+    current_app.logger.info('Retrieving all states')
+    data = CoreData.query.join(Batch).join(State).all()
+    # import pdb; pdb.set_trace()
```
Comment (aslvrstn) [inline app/api/data.py:36]: Can you drop the comment?

**Comment by joshuaellinger** [comment]
Show we rename /data to /core-data to allow for the introduction of /race-data in the future?

File: app/api/data.py (hunk 1 above)
Comment (julia326) [reply app/api/data.py]: done - also, added a test and everything!

**Comment by julia326** [comment]
@joshuaellinger re: renaming /data to /core-data, I think these API endpoint names aren't final yet - but that seems reasonable, can do in a future PR

File: app/api/data.py [hunk 2]
```diff
@@ -21,6 +21,22 @@ def get_batches():
         'batches': [batch.to_dict() for batch in batches]
     })
 
+@api.route('/data/state/', methods=['GET'])
+def get_states():
+    current_app.logger.info('Retrieving all states')
```
Comment (aslvrstn) [inline app/api/data.py:26]: Are these log statements telling us that request logs wouldn't?

File: app/models/data.py [hunk 3]
```diff
@@ -7,25 +7,91 @@
 from datetime import datetime
 from app import db
 
+from sqlalchemy.ext.declarative import declared_attr
+from sqlalchemy.inspection import inspect
+from sqlalchemy.orm import relationship
 
-class Batch(db.Model):
+class DataMixin(object):
```
Comment (aslvrstn) [inline app/models/data.py:14]: Can you comment what this class does? I'm lost

File: app/models/data.py [hunk 4]
```diff
@@ -7,25 +7,91 @@
...
+
+    @declared_attr
+    def __tablename__(cls):
+        return cls.__name__.lower()
+
+    def to_dict(self):
+        d = {}
```
Comment (aslvrstn) [inline app/models/data.py:21]: Is `return {col.name: str(getattr(self, col.name)) for col in self.__table_.columns}` more readable? ...no, right?

**Pull request merged**